frontend/src/components/Auth/Login/Login.js

```diff
@@ -0,0 +1,75 @@
+import "../Auth.scss";
+import React, { useEffect, useState } from "react";
+import TextField from "@material-ui/core/TextField";
+import Button from "@material-ui/core/Button";
+import { Link } from "react-router-dom";
+import useForm from "../../../Hooks/useForm";
+
+const onSubmit = event => {
+  event.preventDefault();
+};
+
+const Login = () => {
+  const [form, setForm] = useForm({
+    email: {
+      invalid: true,
+      touched: false,
+      value: ""
+    },
+    password: {
+      invalid: true,
+      touched: false,
+      value: ""
+    }
+  });
+
+  const [isFormValid, setFormValid] = useState(false);
```

Comment:
Instead of validating it in the component file, how about you run the validation in the hook? You will probably want to validate the passwords/emails in the same way across all components anyway. And after the validation you can make the hook return the errors from validation in the form of an object like : `const errors = {email: "Invalid domain", password: null}`.

So the hook would look like `const [formState, setFormState, formErrors] = useForm()`. And then you can check for existence of any errors and determine based on that if the form is valid or not.